src/writing-your-own-application/generate-project/esp-idf-template.md

```diff
@@ -0,0 +1,185 @@
+# Understanding esp-idf-template
+
+Now that we know how to [generate a std project], let's inspect what the generated project contains and try to understand every part of it.
+
+## Inspecting the generated Project
+
+When creating a project from [esp-idf-template] with the following answers:
+- Which MCU to target? · `esp32c3`
+- Use template default values? · `true`
+
+For this explanation we will use the default values, if you want further modifications, see the [addtional prompts] when not using default values.
+
+It should generate a file structure like this:
+
+```text
+├── .cargo
+│   └── config.toml
+└── src
+    └── main.rs
+├── .gitignore
+├── build.rs
+├── Cargo.toml
+├── rust-toolchain.toml
+├── sdkconfig.defaults
+```
+
+Before going further let's see what these files are for.
+
+- [.cargo/config.toml]
+    - The Cargo configuration
+    - Contains our target
+    - Contains `runner = "espflash flash --monitor"` - this means you can just use `cargo run` to flash and monitor your code
+    - Contains the linker to use, in our case, [`ldproxy`]
+    - Contains the unstable `build-std` cargo feature enabled.
+    - Contains the `ESP-IDF-VERSION` envrionment variable that tells [`esp-idf-sys`] which ESP-IDF version the project will use.
+- src/main.rs
+    - The main source file of the newly created project
+    - We will examine its content in the next section
+- [.gitignore]
+    - Tells `git` which folders and files to ignore
+- [build.rs]
+    - Propagates linker arguments for `ldproxy`.
+- [Cargo.toml]
+    - The usual Cargo manifest declaring some meta-data and dependencies of the project
+- [rust-toolchain.toml]
+    - Defines which Rust toolchain to use
+      - The toolchain will be `nightly` or `esp` depending on your target.
+- [sdkconfig.defaults]
+    - Contains the overriden values from the ESP-IDF defaults.
+
+## `main.rs`
+
+```rust,ignore
+use esp_idf_sys as _; // If using the `binstart` feature of `esp-idf-sys`, always keep this module imported
+
+fn main() {
+    // It is necessary to call this function once. Otherwise some patches to the runtime
+    // implemented by esp-idf-sys might not link properly. See https://github.com/esp-rs/esp-idf-template/issues/71
+    esp_idf_sys::link_patches();
+    println!("Hello, world!");
+}
+```
+The first line its an import that defines the esp-idf entry-point when the root crate is a binary crate that defines a main function.
+
+Then, we have an usual main function with few lines on it:
+- A call to `esp_idf_sys::link_patches` function that makes sure that a few patches to the ESP-IDF which are implemented in Rust are linked to the final executable.
+- We print in our console the famous "Hello World!".
+
+## Running the Code
```

Comment:
This section looks very similar to its counterpart in _Understanding esp-template_ (maybe even the same).

Do you think it is better to have it in one place from the maintenance perspective? If the differences are small, maybe it is worth only covering the differences here?

Just an idea, so feel free to ignore it.